controllers/clusteroperator.go

```diff
@@ -0,0 +1,229 @@
+// status condition change.
+type StatusReason string
+
+const (
+	clusterOperatorName = "baremetal"
+
+	// CO Status Condition Types
+	Available   osconfigv1.ClusterStatusConditionType = osconfigv1.OperatorAvailable
+	Processing  osconfigv1.ClusterStatusConditionType = osconfigv1.OperatorProgressing
+	Degraded    osconfigv1.ClusterStatusConditionType = osconfigv1.OperatorDegraded
+	Upgradeable osconfigv1.ClusterStatusConditionType = osconfigv1.OperatorUpgradeable
+	Disabled    osconfigv1.ClusterStatusConditionType = "Disabled"
+
+	// CO Status Condition Status
+	True    osconfigv1.ConditionStatus = osconfigv1.ConditionTrue
+	False   osconfigv1.ConditionStatus = osconfigv1.ConditionFalse
+	Unknown osconfigv1.ConditionStatus = osconfigv1.ConditionUnknown
+
+	// The default set of status change reasons.
+	ReasonEmpty       StatusReason = ""
+	ReasonComplete    StatusReason = "DeployComplete"
+	ReasonSyncing     StatusReason = "SyncingResources"
+	ReasonSyncFailed  StatusReason = "SyncingFailed"
+	ReasonUnsupported StatusReason = "UnsupportedPlatform"
+)
+
+// defaultStatusConditions returns the default set of status conditions for the
+// ClusterOperator resource used on first creation of the ClusterOperator.
+func defaultStatusConditions(co *osconfigv1.ClusterOperator) []osconfigv1.ClusterOperatorStatusCondition {
+	return []osconfigv1.ClusterOperatorStatusCondition{
+		setStatusCondition(Progressing, False, "", ""),
+		setStatusCondition(Degraded, False, "", ""),
+		setStatusCondition(Available, False, "", ""),
+		setStatusCondition(Disabled, False, "", ""),
+		setStatusCondition(Upgradeable, True, "", ""),
+	}
+}
+
+// relatedObjects returns the current list of ObjectReference's for the
+// ClusterOperator objects's status.
+func (r *ProvisioningReconciler) relatedObjects() []osconfigv1.ObjectReference {
+	return []osconfigv1.ObjectReference{
+		{
+			Group:    "",
+			Resource: "namespaces",
+			Name:     r.config.TargetNamespace,
+		},
+	}
+}
+
+// createClusterOperator creates the ClusterOperator and updates its status.
+func (r *ProvisioningReconciler) createClusterOperator() (*osconfigv1.ClusterOperator, error) {
+	defaultCO := &osconfigv1.ClusterOperator{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterOperator",
+			APIVersion: "config.openshift.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterOperatorName,
+		},
+		Status: osconfigv1.ClusterOperatorStatus{
+			Conditions:     defaultStatusConditions(),
+			RelatedObjects: relatedObjects(),
+			Version:        osconfigv1.OperandVersion{Name: "operator", Version: os.Getenv("OPERATOR_VERSION")},
+		},
+	}
+
+	co, err := r.client.ConfigV1().ClusterOperators().Create(context.Background(), defaultCO, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	co.Status = defaultCO.Status
+
+	return r.client.ConfigV1().ClusterOperators().UpdateStatus(context.Background(), co, metav1.UpdateOptions{})
```

Comment:
This line uses `r.client` but line 100 has `r.osClient` also accessing the cluster operator type. Are the two clients the same? Do we need both?

Reply:
Just go all the make targets to compile correctly. There were a lot of errors. Still iterating, but this particular one should be resolved.